src/transformers/pipelines/base.py

```diff
@@ -793,11 +793,16 @@ def __init__(
             if isinstance(device, torch.device):
                 self.device = device
             elif isinstance(device, str):
+                if device == "cuda":
```

Comment:
Maybe?
```suggestion
                if device == "cuda" and not hasattr(torch.device, "__enter__"):
```

Reply:
Just double checking, if this condition is true, does the line below run OK?

`self.device = torch.device(device)`